src/Storages/IRemoteFileMetadata.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+#include <functional>
+#include <memory>
+#include <unordered_map>
+#include <base/types.h>
+#include <boost/core/noncopyable.hpp>
+
+namespace DB
+{
+class IRemoteFileMetadata
```

Comment:
Lets also move this file to `Storages/Cache`

Reply:
done